core/src/main/scala/org/apache/spark/status/protobuf/ApplicationEnvironmentInfoWrapperSerializer.scala

```diff
@@ -23,8 +23,14 @@ import org.apache.spark.resource.{ExecutorResourceRequest, TaskResourceRequest}
 import org.apache.spark.status.ApplicationEnvironmentInfoWrapper
 import org.apache.spark.status.api.v1.{ApplicationEnvironmentInfo, ResourceProfileInfo, RuntimeInfo}
 
-object ApplicationEnvironmentInfoWrapperSerializer {
-  def serialize(input: ApplicationEnvironmentInfoWrapper): Array[Byte] = {
+class ApplicationEnvironmentInfoWrapperSerializer extends ProtoBufSerDe {
+
+  override val supportClass: Class[_] = classOf[ApplicationEnvironmentInfoWrapper]
+
+  override def serialize(input: Any): Array[Byte] =
```

Comment:
Is it possible to change the `Any` here to be `ApplicationEnvironmentInfoWrapper`? The trait `ProtoBufSerDe` can become `ProtoBufSerDe[T]`.
I haven't look into it closely. But if we can avoid the `asInstanceOf` here, the proposal looks good.

Reply:
I'm not sure whether SPI can support generic type, but let me try